### PR TITLE
Manage client roles for users

### PR DESCRIPTION
**Pull request by HairyMike**
This incorporates the work done in https://github.com/Nerzal/gocloak/pull/76 and adds some tests.

I'm creating a new PR in the hope that this can go to master as https://github.com/Nerzal/gocloak/pull/76 hasn't been updated in a number of months.

### CODE REVIEW

**Comment by SVilgelm** [comment]
@Nerzal don't merge this, I still don't understand the reason of failed tests

**Comment by HairyMike** [comment]
Haven't looked at this for a few weeks.  I think some tests might be failing on master, possibly due to keycloak changes.  I'll get into this again soon (since we need this functionality).

**Comment by SVilgelm** [comment]
I already fixed master, only this PR fails now

**Comment by HairyMike** [comment]
looks like something up with the realm file:
```
Unable to import realm gocloak from file /tmp/gocloak-realm.json.: java.lang.RuntimeException: Unable to find composite client role: query-clients
```

File: testdata/gocloak-realm.json [hunk 1]
```diff
@@ -369,6 +369,15 @@
         "phone",
         "offline_access"
       ]
+    },
+    {
+      "id": "6631a5c9-20b8-45aa-a271-eb6e8c8c4c99",
```
Comment (SVilgelm) [inline testdata/gocloak-realm.json]: seems, it's a wrong place for this role

Comment (HairyMike) [reply testdata/gocloak-realm.json]: on it, thanks

**Comment by SVilgelm** [comment]
> 14:06:34,370 WARN  [org.keycloak.services] (ServerService Thread Pool -- 68) KC-SERVICES0005: Unable to import realm gocloak from file /tmp/gocloak-realm.json.: org.keycloak.models.ModelDuplicateException: javax.persistence.EntityExistsException: A different object with the same identifier value was already associated with the session : [org.keycloak.models.jpa.entities.RoleEntity#7e0a25f0-5662-43e1-898c-646eb98dbf4d]

**Comment by SVilgelm** [comment]
> 08:24:19,778 WARN  [org.keycloak.services] (ServerService Thread Pool -- 65) KC-SERVICES0005: Unable to import realm gocloak from file /tmp/gocloak-realm.json.: java.lang.RuntimeException: Unable to find composite client role: query-clients

**Comment by SVilgelm** [comment]
Close due to #105

**Pull request closed without merging**